Gems/O3DE/GeomNodes/Code/Source/Editor/Rendering/GNRenderMesh.cpp

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "GNRenderMesh.h"
+
+#include <Atom/RPI.Public/Model/Model.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Reflect/Asset/AssetUtils.h>
+#include <Atom/RPI.Reflect/Model/ModelAssetCreator.h>
+#include <Atom/RPI.Reflect/Model/ModelLodAssetCreator.h>
+#include <Atom/RPI.Reflect/ResourcePoolAssetCreator.h>
+#include <AtomLyIntegration/CommonFeatures/Material/MaterialComponentBus.h>
+#include <AzCore/Math/PackedVector3.h>
+
+namespace GeomNodes
+{
+    GNRenderMesh::GNRenderMesh(AZ::EntityId entityId)
+        : m_entityId(entityId)
+    {
+        AZ::Render::MeshHandleStateRequestBus::Handler::BusConnect(m_entityId);
+        AZ::Render::MaterialConsumerRequestBus::Handler::BusConnect(m_entityId);
+        AZ::Render::MaterialComponentNotificationBus::Handler::BusConnect(m_entityId);
+    }
+
+    GNRenderMesh::~GNRenderMesh()
+    {
+        if (m_meshHandle.IsValid() && m_meshFeatureProcessor)
+        {
+            m_meshFeatureProcessor->ReleaseMesh(m_meshHandle);
+            AZ::Render::MeshHandleStateNotificationBus::Event(
+                m_entityId, &AZ::Render::MeshHandleStateNotificationBus::Events::OnMeshHandleSet, &m_meshHandle);
+        }
+
+		AZ::Render::MaterialConsumerRequestBus::Handler::BusDisconnect();
+		AZ::Render::MaterialComponentNotificationBus::Handler::BusDisconnect();
+        AZ::Render::MeshHandleStateRequestBus::Handler::BusDisconnect();
+        //AZ::TickBus::Handler::BusDisconnect();
```

Comment:
remove if not needed

Reply:
removed in commit [d756b38](https://github.com/o3de/o3de-technicalart/pull/11/commits/d756b38541172fd3e0051140ca96cc9bd8b05388)